driver/image_adjust_brightness_driver.hpp

```diff
@@ -0,0 +1,301 @@
+
+    uint32_t ctx = 0;
+
+    input_gpu  = std::unique_ptr<GPUMem>(new GPUMem(ctx, input_size, sizeof(Tgpu)));
+    output_gpu = std::unique_ptr<GPUMem>(new GPUMem(ctx, output_size, sizeof(Tgpu)));
+
+    in_host  = std::vector<Tgpu>(input_size, static_cast<Tgpu>(0));
+    out_host = std::vector<Tgpu>(output_size, static_cast<Tgpu>(0));
+    out_ref  = std::vector<Tref>(output_size, static_cast<Tref>(0));
+
+    for(auto i = 0; i < input_size; i++)
+    {
+        in_host[i] = static_cast<Tgpu>(i);
+    }
+
+    if(input_gpu->ToGPU(GetStream(), in_host.data()) != miopenStatusSuccess)
+        std::cerr << "Error copying buffer to GPU with size " << input_gpu->GetSize() << std::endl;
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ImageAdjustBrightnessDriver<Tgpu, Tref>::RunForwardGPU()
+{
+    float kernel_total_time = 0;
+    float kernel_first_time = 0;
+
+    Timer t;
+    START_TIME
+
+    for(int i = 0; i < inflags.GetValueInt("iter"); i++)
+    {
+        miopenImageAdjustBrightness(GetHandle(),
+                                    inputTensorDesc,
+                                    outputTensorDesc,
+                                    input_gpu->GetMem(),
+                                    output_gpu->GetMem(),
+                                    brightness_factor);
+
+        float time = 0.0;
+        miopenGetKernelTime(GetHandle(), &time);
+        kernel_total_time += time;
+        if(i == 0)
+            kernel_first_time = time;
+    }
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        STOP_TIME
+        int iter = inflags.GetValueInt("iter");
+
+        if(WALL_CLOCK)
+            std::cout << "Image Adjust Brightness Time Elapsed: " << kernel_total_time << " ms"
+                      << std::endl;
+
+        float kernel_avg_time =
+            iter > 1 ? (kernel_total_time - kernel_first_time) / (iter - 1) : kernel_total_time;
+
+        std::cout << "Image Adjust Brightness Kernel Time Elapsed: " << kernel_avg_time << " ms"
+                  << std::endl;
+    }
+
+    if(output_gpu->FromGPU(GetStream(), out_host.data()) != miopenStatusSuccess)
+    {
+        std::cerr << "Error copying buffer from GPU with size " << output_gpu->GetSize()
+                  << std::endl;
+        return -1;
+    }
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ImageAdjustBrightnessDriver<Tgpu, Tref>::RunForwardCPU()
+{
+    mloImageAdjustBrightnessRunHost(in_host.data(),
+                                    out_ref.data(),
+                                    miopen::deref(inputTensorDesc),
+                                    miopen::deref(outputTensorDesc),
+                                    brightness_factor);
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int ImageAdjustBrightnessDriver<Tgpu, Tref>::VerifyForward()
+{
+    RunForwardCPU();
+
+    auto threashold = sizeof(Tgpu) == 4 ? 1e-6 : 5e-2;
```

Comment:
Is there any reason you use these constants as threasholds? I often use 
```
std::numeric_limits<Tgpu>::epsilon()
```

Reply:
Check the other drivers for details. It's seems acceptable to use custom values here if desired.

Reply:
Is it oke @et16kr

Reply:
As far as I understand you *can* set your own threashold (check `pool_driver`). I've changed to `numeric_limits::epsilon`